examples/applyconfig/api/v1/ac/zz_generated.applyconfigurations.go

```diff
@@ -0,0 +1,49 @@
+// +build !ignore_autogenerated
+
+// Code generated by controller-gen. DO NOT EDIT.
+
+package v1
+
+import (
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	metav1 "k8s.io/client-go/applyconfigurations/meta/v1"
+	apiv1 "sigs.k8s.io/controller-runtime/examples/applyconfig/api/v1"
+)
+
+// FooApplyConfiguration represents a declarative configuration of the Foo type for use
+// with apply.
+type FooApplyConfiguration struct {
+	metav1.TypeMetaApplyConfiguration    `json:",inline"`
+	*metav1.ObjectMetaApplyConfiguration `json:"metadata,omitempty"`
+	NonNullableField                     *string `json:"nonNullableField,omitempty"`
+	NullableField                        *string `json:"nullableField,omitempty"`
+}
+
+// WithNonNullableField sets the NonNullableField field in the declarative configuration to the given value
+func (b *FooApplyConfiguration) WithNonNullableField(value string) *FooApplyConfiguration {
+	b.NonNullableField = &value
+	return b
+}
+
+// WithNullableField sets the NullableField field in the declarative configuration to the given value
+func (b *FooApplyConfiguration) WithNullableField(value string) *FooApplyConfiguration {
+	b.NullableField = &value
+	return b
+}
+
+// FooApplyConfiguration represents a declarative configuration of the Foo type for use
+// with apply.
+func Foo() *FooApplyConfiguration {
+	return &FooApplyConfiguration{}
+}
+func (ac *FooApplyConfiguration) GetReferenceType() runtime.Object {
```

Comment:
Can we maybe include this method when dealing with appyconfigs in an interface, so that we end up with an interface that is not fulfilled by `metav1.Object`?